projects/ngx-maplibre-gl/src/lib/source/geojson/geojson-source.component.ts

```diff
@@ -38,81 +39,89 @@ import { MapService } from '../../map/map.service';
+  readonly attribution = input<GeoJSONSourceSpecification['attribution']>();
+
   /** Dynamic input */
-  @Input() clusterMinPoints?: GeoJSONSourceSpecification['clusterMinPoints'];
+  readonly buffer = input<GeoJSONSourceSpecification['buffer']>();
+
   /** Dynamic input */
-  @Input() clusterProperties?: GeoJSONSourceSpecification['clusterProperties'];
+  readonly tolerance = input<GeoJSONSourceSpecification['tolerance']>();
+
   /** Dynamic input */
-  @Input() lineMetrics?: GeoJSONSourceSpecification['lineMetrics'];
+  readonly cluster = input<GeoJSONSourceSpecification['cluster']>();
+
   /** Dynamic input */
-  @Input() generateId?: GeoJSONSourceSpecification['generateId'];
+  readonly clusterRadius = input<GeoJSONSourceSpecification['clusterRadius']>();
+
   /** Dynamic input */
-  @Input() promoteId?: GeoJSONSourceSpecification['promoteId'];
+  readonly clusterMaxZoom =
+    input<GeoJSONSourceSpecification['clusterMaxZoom']>();
+
   /** Dynamic input */
-  @Input() filter?: GeoJSONSourceSpecification['filter'];
+  readonly clusterMinPoints =
+    input<GeoJSONSourceSpecification['clusterMinPoints']>();
 
-  /** @hidden */
-  type: GeoJSONSourceSpecification['type'] = 'geojson';
+  /** Dynamic input */
+  readonly clusterProperties =
+    input<GeoJSONSourceSpecification['clusterProperties']>();
 
-  updateFeatureData = new Subject();
+  /** Dynamic input */
+  readonly lineMetrics = input<GeoJSONSourceSpecification['lineMetrics']>();
 
-  private sub = new Subscription();
-  private sourceAdded = false;
-  private featureIdCounter = 0;
+  /** Dynamic input */
+  readonly generateId = input<GeoJSONSourceSpecification['generateId']>();
 
-  constructor(private mapService: MapService, private zone: NgZone) {}
+  /** Dynamic input */
+  readonly promoteId = input<GeoJSONSourceSpecification['promoteId']>();
 
-  ngOnInit() {
-    if (!this.data) {
-      this.data = {
-        type: 'FeatureCollection',
-        features: [],
-      };
-    }
-    const sub1 = this.mapService.mapLoaded$.subscribe(() => {
-      this.init();
-      const sub = fromEvent(this.mapService.mapInstance, 'styledata')
-        .pipe(filter(() => !this.mapService.mapInstance.getSource(this.id)))
-        .subscribe(() => {
-          this.init();
-        });
-      this.sub.add(sub);
-    });
-    this.sub.add(sub1);
+  /** Dynamic input */
+  readonly filter = input<GeoJSONSourceSpecification['filter']>();
+
+  private readonly updateFeatureDataSubject = new Subject<void>();
+
+  private readonly featureIdCounter = signal(0);
+
+  constructor() {
+    this.sourceDirective.loadSource$
+      .pipe(
+        tap(() =>
+          this.sourceDirective.addSource(this.getGeoJSONSourceSpecification())
+        ),
+        switchMap(() => this.updateFeature()),
+        takeUntilDestroyed()
+      )
+      .subscribe();
   }
 
   ngOnChanges(changes: SimpleChanges) {
-    if (!this.sourceAdded) {
+    if (!this.sourceDirective.sourceId()) {
```

Comment:
Are we sure about the removal of "sourceAdded" variable?

Reply:
This was moved to the directive, I see, maybe a different name like `SourceDirectiveHelper` can convey what it does better?

Reply:
I'm not very convinced to name it as is for two reasons: 

The first one: it is not really a helper
The second: Angular convention means that a directive must end with "Directive" like ExampleDirective.

Reply:
Fair enough.